source/tokens/Token.hx

```diff
@@ -0,0 +1,22 @@
+package tokens;
+
+import flixel.FlxBasic;
+import flixel.util.FlxColor;
+import obstacle.Obstacle;
+
+class Token extends Obstacle
+{
+	override function init(parameters:ObstacleParameters)
+	{
+		super.init(parameters);
+		color = FlxColor.YELLOW;
+	}
+
+	public static function overlapsWithPlayer(player:FlxBasic, scoreToken:Token)
+	{
+		scoreToken.kill();  //Remove token when collide with player
+		//var hud:Hud;
+		//hud.addScore(1);    //Add 1 to coin score when player collides
```

Comment:
This code has an error because declaring a `Hud` variable doesn't populate it with any kind of value - it's just containing `null`, which can't handle being told to execute `.addScore(1)`

To make it work, each instance of `Token` needs access to the `Hud` object that we created inside of `PlayState`. There are a bunch of ways to do this - you could have the constructor take in a `Hud` object that it can use when it's time to call `addScore()`

Hud does something similar - it [takes a Player object as part of its constructor](https://github.com/SamBumgardner/csc-303-game-2022/blob/main/source/Hud.hx#L24-L28) which it uses later when it wants to [update the current health text](https://github.com/SamBumgardner/csc-303-game-2022/blob/main/source/Hud.hx#L82-L85)

Alternatively you could do something like make [the `hud` field in the `PlayState`](https://github.com/SamBumgardner/csc-303-game-2022/blob/main/source/PlayState.hx#L18) public, then use `FlxG.state` to get a reference to our current state and [cast it to the PlayState type](https://haxe.org/manual/expression-cast.html). It could end up looking like 
```
cast (FlxG.state, PlayState).hud.addScore(1);
```